manifests/init.pp

```diff
@@ -7,69 +7,123 @@
+  String[1]                          $poudriere_data             = '${BASEFS}/data',
+  Enum['yes', 'no']                  $use_portlint               = 'no',
+  Optional[String[1]]                $mfssize                    = undef,
+  String[1]                          $tmpfs                      = 'yes',
+  Optional[Integer[1]]               $tmpfs_limit                = undef,
+  Optional[Integer[1]]               $max_memory                 = undef,
+  Optional[Integer[1]]               $max_files                  = undef,
+  Stdlib::Absolutepath               $distfiles_cache            = '/usr/ports/distfiles',
+  Optional[String[1]]                $svn_host                   = undef,
+  Enum['yes', 'no', 'verbose']       $check_changed_options      = 'verbose',
+  Enum['yes', 'no']                  $check_changed_deps         = 'yes',
+  Optional[Enum['yes', 'no']]        $bad_pkgname_deps_are_fatal = undef,
+  Optional[String[1]]                $pkg_repo_signing_key       = undef,
+  Boolean                            $ccache_enable              = false,
+  Stdlib::Absolutepath               $ccache_dir                 = '/var/cache/ccache',
+  Optional[Stdlib::Absolutepath]     $ccache_static_prefix       = undef,
+  Optional[Enum['yes', 'no']]        $restrict_networking        = undef,
+  Optional[String[1]]                $allow_networking_packages  = undef,
+  Integer[1]                         $parallel_jobs              = $facts['processors']['count'],
+  Optional[Integer[1]]               $prepare_parallel_jobs      = undef,
+  Optional[String[1]]                $save_wrkdir                = undef,
+  Optional[String[1]]                $wrkdir_archive_format      = undef,
+  Optional[String[1]]                $nolinux                    = undef,
+  Optional[Enum['yes', 'no']]        $no_force_package           = undef,
+  Optional[String[1]]                $no_package_building        = undef,
+  Optional[String[1]]                $http_proxy                 = undef,
+  Optional[String[1]]                $ftp_proxy                  = undef,
+  Optional[String[1]]                $no_restricted              = undef,
+  Optional[String[1]]                $allow_make_jobs            = undef,
+  Optional[String[1]]                $allow_make_jobs_packages   = undef,
+  Optional[Enum['yes', 'no']]        $timestamp_logs             = undef,
+  Optional[String[1]]                $url_base                   = undef,
+  Optional[Integer[1]]               $max_execution_time         = undef,
+  Optional[Integer[1]]               $nohang_time                = undef,
+  Optional[Enum['yes', 'no']]        $atomic_package_repository  = undef,
+  Optional[Enum['yes', 'no']]        $commit_packages_on_failure = undef,
+  Optional[Enum['yes', 'no']]        $keep_old_packages          = undef,
+  Optional[Integer]                  $keep_old_packages_count    = undef,
```

Comment:
what are possible values here? can we enforce the lower (and upper) limit of the integer?

Reply:
Yes, we can enforce a minimum.  I don't see a reason for an arbitrary max though.